src/cron-jobs/cron-jobs.service.ts

```diff
@@ -0,0 +1,28 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { Cron, CronExpression } from '@nestjs/schedule';
+import { LogoutTokenRepository } from '../auth/logout-token.repository';
+import { errorMessages } from '../constants/error-messages';
+
+@Injectable()
+export class CronJobsService {
+  constructor(private readonly logoutTokenRepository: LogoutTokenRepository) {}
+
+  @Cron(CronExpression.EVERY_DAY_AT_3PM)
+  async deleteLogoutTokens() {
+    const tokens = await this.logoutTokenRepository.find();
+
+    tokens.map((token) => {
+      const { created_at } = token;
+      const week = 7 * 24 * 60 * 60 * 1000;
+      const tokenTimeCreation = created_at.getTime();
+
+      if (Date.now() - tokenTimeCreation > week) {
+        try {
+          this.logoutTokenRepository.delete(token.id);
+        } catch (error) {
+          throw new BadRequestException(errorMessages.DELETE_OLD_LOGOUT_TOKENS_ERROR);
+        }
+      }
+    });
```

Comment:
1. The number of tokens can be large, so it’s better not to iterate over them in a map loop, but to immediately select tokens from the database with the required creation date. Then delete all of them.
2. The operation of deleting from the database is asynchronous, so to get correct results and correct error handling, we need to wait for its execution (await), the .map() array method does not wait for awaits.
3. The line "throw new BadRequestException(errorMessages.DELETE_OLD_LOGOUT_TOKENS_ERROR);" does not make sense in the context of this task. We do not receive any requests, and there is no point in sending BadRequestException anywhere. Also, in the current implementation, there may be more than one error. Logging the error will be better.